### PR TITLE
feature: created new component : spinning text

### PR DESCRIPTION
**Pull request by ibtihelbs**

### CODE REVIEW

File: components/core/spinning-text.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
+'use client';
+import { motion } from 'framer-motion';
+import React, { CSSProperties, ReactElement } from 'react';
+
+// Define the props types for the SpinningText component
+interface SpinningTextProps {
+  children: ReactElement;
+  width?: number;
+  textStyle?: CSSProperties;
+  degree?: number;
```
Comment (ibelick) [inline components/core/spinning-text.tsx:10]: can we add props:
- className
- duration

File: components/core/spinning-text.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+export const SpinningText: React.FC<SpinningTextProps> = ({
+  children,
+  width = 200,
+  textStyle = {},
+  degree = 180,
+}) => {
+  const textEl = children.props.children || children.props.content || '';
+  const toArr = typeof textEl === 'string' ? textEl.split('') : [];
+  console.log(toArr);
```
Comment (ibelick) [inline components/core/spinning-text.tsx:21]: remove log

File: components/core/spinning-text.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  const fraction = degree / toArr.length;
+
+  return (
+    <motion.div
+      className='border-3 relative flex items-center justify-center rounded-full border-current'
```
Comment (ibelick) [inline components/core/spinning-text.tsx:26]: remove `border-3`

File: components/core/spinning-text.tsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+      style={{
+        height: `${width}px`,
+        width: `${width}px`,
+      }}
+      animate={{ rotate: 360 }} // Rotate the entire circle
```
Comment (ibelick) [inline components/core/spinning-text.tsx:31]: remove comments please (`// Rotate the entire circle` and others)

File: components/core/spinning-text.tsx (hunk 1 above)
Comment (ibtihelbs) [reply components/core/spinning-text.tsx:9]: I added both duration and className.

File: components/core/spinning-text.tsx (hunk 4 above)
Comment (ibtihelbs) [reply components/core/spinning-text.tsx]: done

File: components/core/spinning-text.tsx (hunk 2 above)
Comment (ibtihelbs) [reply components/core/spinning-text.tsx]: done!

**Comment by ibtihelbs** [comment]
Changes are : Duration: Added a duration prop to control the speed of the rotation.
ClassName: Added a className prop for applying custom styles.
Text Extraction: Adjusted text extraction to handle both direct text and components with a content prop.
Default Values: Set default values for duration and className, with className defaulting to any class from the child component if provided.

**Comment by ibtihelbs** [comment]
hope you check the new changes!

**Comment by ibelick** [comment]
@ibtihelbs I'll take a look soon. First I need to better organize the motion-primitives components

**Comment by ibelick** [comment]
Thanks for your contribution @ibtihelbs! Cf https://github.com/ibelick/motion-primitives/pull/68

**Pull request closed without merging**